rust/src/lib/ifaces/bond.rs

```diff
@@ -0,0 +1,475 @@
+    }
+
+    pub(crate) fn validate(
+        &self,
+        base: &BaseInterface,
+    ) -> Result<(), NmstateError> {
+        if let Some(opts) = &self.options {
+            if let Some(mode) = &self.mode {
+                opts.validate(mode, base)?;
+            } else {
+                return Err(NmstateError::new(
+                    ErrorKind::ValueError,
+                    "Bond mode is mandatory".to_string(),
+                ));
+            }
+        }
+        Ok(())
+    }
+
+    pub(crate) fn update(&mut self, other: Option<&BondConfig>) {
+        if let Some(other) = other {
+            self.mode = other.mode.clone();
+            self.options = other.options.clone();
+            self.port = other.port.clone();
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondAdSelect {
+    #[serde(alias = "0")]
+    Stable,
+    #[serde(alias = "1")]
+    Bandwidth,
+    #[serde(alias = "2")]
+    Count,
+}
+
+impl BondAdSelect {
+    pub fn to_u8(&self) -> u8 {
+        match self {
+            Self::Stable => 0,
+            Self::Bandwidth => 1,
+            Self::Count => 2,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondLacpRate {
+    #[serde(alias = "0")]
+    Slow,
+    #[serde(alias = "1")]
+    Fast,
+}
+
+impl BondLacpRate {
+    pub fn to_u8(&self) -> u8 {
+        match self {
+            Self::Slow => 0,
+            Self::Fast => 1,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondAllPortsActive {
+    #[serde(alias = "0")]
+    Dropped,
+    #[serde(alias = "1")]
+    Delivered,
+}
+
+impl BondAllPortsActive {
+    pub fn to_u8(&self) -> u8 {
+        match self {
+            Self::Dropped => 0,
+            Self::Delivered => 1,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondArpAllTargets {
+    #[serde(alias = "0")]
+    Any,
+    #[serde(alias = "1")]
+    All,
+}
+
+impl BondArpAllTargets {
+    pub fn to_u32(&self) -> u32 {
+        match self {
+            Self::Any => 0,
+            Self::All => 1,
+        }
+    }
+}
+
+const BOND_STATE_ACTIVE: u8 = 0;
+const BOND_STATE_BACKUP: u8 = 1;
+
+const BOND_ARP_VALIDATE_NONE: u32 = 0;
+const BOND_ARP_VALIDATE_ACTIVE: u32 = 1 << BOND_STATE_ACTIVE as u32;
+const BOND_ARP_VALIDATE_BACKUP: u32 = 1 << BOND_STATE_BACKUP as u32;
+const BOND_ARP_VALIDATE_ALL: u32 =
+    BOND_ARP_VALIDATE_ACTIVE | BOND_ARP_VALIDATE_BACKUP;
+const BOND_ARP_FILTER: u32 = BOND_ARP_VALIDATE_ALL + 1;
+const BOND_ARP_FILTER_ACTIVE: u32 = BOND_ARP_VALIDATE_ACTIVE | BOND_ARP_FILTER;
+const BOND_ARP_FILTER_BACKUP: u32 = BOND_ARP_VALIDATE_BACKUP | BOND_ARP_FILTER;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```

Comment:
`#[serde(rename_all = "kebab-case")]` is better.

Reply:
Done!